presto-main/src/main/java/com/facebook/presto/transaction/InMemoryTransactionManager.java

```diff
@@ -174,6 +174,25 @@ public List<TransactionInfo> getAllTransactionInfos()
                 .collect(toImmutableList());
     }
 
+    @Override
+    public void tryRegisterTransaction(TransactionInfo transactionInfo)
+    {
+        TransactionId transactionId = transactionInfo.getTransactionId();
+        if (transactions.containsKey(transactionId)) {
+            return;
+        }
+        BoundedExecutor executor = new BoundedExecutor(finishingExecutor, maxFinishingConcurrency);
+        TransactionMetadata transactionMetadata = new TransactionMetadata(transactionId,
+                transactionInfo.getIsolationLevel(),
+                transactionInfo.isReadOnly(),
+                transactionInfo.isAutoCommitContext(),
```

Comment:
consider extracting this part to a private helper function and reuse it in tryRegisterTransaction and beginTransaction.